redbeams/src/main/java/com/sequenceiq/redbeams/service/network/AwsSubnetValidator.java

```diff
@@ -0,0 +1,30 @@
+package com.sequenceiq.redbeams.service.network;
+
+import java.util.Collection;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudSubnet;
+import com.sequenceiq.redbeams.exception.BadRequestException;
+
+@Component
+public class AwsSubnetValidator {
+
+    public void validate(Collection<CloudSubnet> cloudSubnets, int minSubnetNeededInDifferentAz) {
+        validateEnoughSubnetIsPresent(cloudSubnets, minSubnetNeededInDifferentAz);
+        validateAzNumber(cloudSubnets, minSubnetNeededInDifferentAz);
+    }
+
+    private void validateAzNumber(Collection<CloudSubnet> cloudSubnets, int minSubnetNeededInDifferentAz) {
+        long numAZs = cloudSubnets.stream().map(CloudSubnet::getAvailabilityZone).distinct().count();
+        if (numAZs < minSubnetNeededInDifferentAz) {
+            throw new BadRequestException("All subnets are in the same availability zone: at least two subnets in two different availability zones needed");
+        }
+    }
+
+    private void validateEnoughSubnetIsPresent(Collection<CloudSubnet> cloudSubnets, int minSubnetNeededInDifferentAz) {
+        if (cloudSubnets.size() < minSubnetNeededInDifferentAz) {
+            throw new BadRequestException("Insufficient number of subnets: at least two subnets in two different availability zones needed");
```

Comment:
In a generalized solution this message is not valid any more. I would suggest "Insufficient number of subnets: at least %d subnets required"